workers_test.go

```diff
@@ -384,13 +407,33 @@ func TestWorkers_DownloadWorkerWithNameErrorsWithoutAccountId(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestWorkers_DownloadWorkerModule(t *testing.T) {
+	setup(Debug(true),UsingAccount("foo"))
+	defer teardown()
+
+	mux.HandleFunc("/accounts/foo/workers/scripts/bar", func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, http.MethodGet, r.Method, "Expected method 'GET', got %s", r.Method)
+		w.Header().Set("content-type", "multipart/form-data; boundary=workermoduledownloadresponse")
+		fmt.Fprintf(w, workerModuleDownloadResponse) //nolint
+	})
+	res, err := client.DownloadWorker(context.Background(), &WorkerRequestParams{ScriptName: "bar"})
+	want := WorkerScriptResponse{
+		successResponse,
+		WorkerScript{
+			Script: workerModuleScript,
+		}}
+	if assert.NoError(t, err) {
+		assert.Equal(t, want.Script, res.Script)
+	}
+}
+
 func TestWorkers_ListWorkerScripts(t *testing.T) {
 	setup(UsingAccount("foo"))
 	defer teardown()
 
 	mux.HandleFunc("/accounts/foo/workers/scripts", func(w http.ResponseWriter, r *http.Request) {
 		assert.Equal(t, http.MethodGet, r.Method, "Expected method 'GET', got %s", r.Method)
-		w.Header().Set("content-type", "application-json")
+		w.Header().Set("content-type", "application/json")
```

Comment:
`makeRequestWithAuthTypeAndHeaders` doesn't look at the content-type of the response, so it never flagged the incorrect `application-json` content-type before. I think it would be helpful to be stricter about only parsing JSON responses if the response content-type is set appropriately.